gpt4all-kotlin/src/jvmMain/kotlin/com/xebia/functional/gpt4all/Models.kt

```diff
@@ -0,0 +1,32 @@
+package com.xebia.functional.gpt4all
+
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.Json
+import java.net.URL
+
+@Serializable
+data class Gpt4AllModel(
+  val order: String,
+  val md5sum: String,
+  val name: String,
+  val filename: String,
+  val filesize: String,
+  val requires: String? = null,
+  val ramrequired: String,
+  val parameters: String,
+  val quant: String,
+  val type: String,
+  val description: String,
+  val disableGUI: String? = null,
+  val url: String? = null
+) {
+  companion object {
+    private val url = "https://raw.githubusercontent.com/nomic-ai/gpt4all/main/gpt4all-chat/metadata/models.json"
+    fun supportedModels(): List<Gpt4AllModel> {
```

Comment:
Is this going to run the HTTP operation and fetch the list of supported models from that `URL` on every call?

Reply:
For now, yes, We have discussed the concept of a Model Zoo / List that we support beyond the GPT4ALL ones with our curated list of models. Feel free to commit any improvements directly to this branch if we want to cache the results. I may not have much more time today to dedicate to it.

Reply:
Let's address this when we have a strategy for our recommended models.